mantidproject/mantid
# Add FileComparisonHelper and rewrite SaveGSS tests

Description of work.
This PR adds the `FileComparisonHelper` found in `SaveOpenGenieAscii` to the Mantid test helpers. It also rewrites SaveGSSTest to use the new unit tests in prep for some future extra options.

---

The `FileComparisonHelper` namespace contains several methods to allow developers to compare if two files are identical for testing small files.
The easiest way to use this new method is to add a reference file to the unit test folder and call `TS_ASSERT(isEqualToReferenceFile("Reference file name", outputFilePath));` and let it handle everything for you. 
The namespace also has additional methods for advanced usage such as handling file streams or buffered char streams directly if you desire.
If two files differ it will return false and log an error showing what character it found and what it expected, where, and the line up till that point to help diagnose why a failure occurred. 

---

**To test:**
Code review
Change a reference file to check that it correctly detects if they differ and gives a sane error message in the unit test.

Fixes - #19413 (We will tidy SaveGSS in a separate PR)

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
